wgengine/magicsock/magicsock_test.go

```diff
@@ -542,6 +543,140 @@ func TestDeviceStartStop(t *testing.T) {
 	dev.Close()
 }
 
+type testConnClosingContext struct {
+	parent context.Context
+	mu     *sync.Mutex
```

Comment:
what's this guard?  oh, this is only for Err?

Reply:
Yes, at present that is a pointer to the conn.mu and is used to deliberately acquire the lock in order to cause a deadlock if the code-under-test tries to grab the lock after calling Err().